CHANGES.rst

```diff
@@ -776,6 +776,10 @@ API changes
     for parsing a file with blank lines in the header containing whitespace will
     need to be updated. [#2654]
 
+  - if any keys in the formats dictionary are not to match any keys in the names
```

Comment:
Sentences must always be capitalized (here and in code comments).  Along the lines that @embray already suggested, try:
```
 - Check that columns in `formats` specifier exist in the output table when writing. [#4508]
```